tooling/helpers/0001-db-migration-add-script-to-squash-DB-migrations.patch

```diff
@@ -0,0 +1,1237 @@
+From 138b178fc7bb1a789f1cf48e367cb0ed9c60cd9c Mon Sep 17 00:00:00 2001
+From: Mark Yen <mark.yen@suse.com>
+Date: Wed, 15 May 2019 12:03:32 -0700
+Subject: [PATCH] db migration: add script to squash DB migrations
+
+This adds a rake task to squash DB migrations, outputting a SCF-style
+patch that will blank out the squashed migration scripts, and instead
+create a new migration that configures the database schema from scratch.
+This should lead to faster migrations (as we don't need to repeatedly
+do schema changes), and let things fail safer (as most migrations now do
+nothing, they can be replayed with no worries).
+
+The migration script has a hard-coded date that is the limit for
+squashing, so that we can still upgrade from existing installations that
+may have some of the migrations applied.
+
+It appears that there are still some differences with a database
+generated via the squashed migration, compared to a normal one:
+
+- Missing / added charset=utf8, collate=utf8_bin
+  (This shouldn't be an issue, because of table defaults)
+- Some indexes &c. are in a different column order.
+  This is not expected to cause any issues.
```

Comment:
I also see changes to the initial auto-increment settings, i.e `CREATE TABLE ( ... ) AUTO_INCREMENT=N`, where N changed. Thought I should note that, despite not expecting these to cause issues either.